backend/graphql/resolvers.ts

```diff
@@ -0,0 +1,166 @@
+import { getAllTracks, getTrack, addTrack, updateTrackById, removeTrack, removeTracks } from '../controllers/tracks.controller';
+import { createSlug } from '../utils/slug';
+
+export const resolvers = {
+  Query: {
+    health: async () => 'ok',
+    genres: async () => {
+      return getGenres();
+    },
+    tracks: async (_parent: any, args: any) => {
+      const params = {
+        page: args.page,
+        limit: args.limit,
+        sort: args.sort,
+        order: args.order,
+        search: args.search,
+        genre: args.genre,
+        artist: args.artist,
+      };
+      const { tracks, total } = await getTracks(params);
+      const page = params.page || 1;
+      const limit = params.limit || 10;
+      return {
+        data: tracks,
+        meta: {
+          total,
+          page,
+          limit,
+          totalPages: Math.ceil(total / limit)
+        }
+      };
+    },
+    track: async (_parent: any, args: { slug: string }, context: any) => {
+      return getTrack({ ...context.request, params: { slug: args.slug } }, context.reply);
+    },
+  },
+  Mutation: {
+    createTrack: async (_parent: any, args: any) => {
+      const { title, artist, album, genres, coverImage } = args.input;
+      if (!title || !artist || !genres) {
+        throw new Error('Title, artist, and genres are required');
+      }
+      const slug = createSlug(title);
+      const newTrack = await createTrack({
+        title,
+        artist,
+        album,
+        genres,
+        coverImage,
+        slug
+      });
+      return newTrack;
+    },
+    updateTrack: async (_parent: any, args: any) => {
+      const { id, input } = args;
+      const existingTrack = await getTrackById(id);
+      if (!existingTrack) {
+        throw new Error('Track not found');
+      }
+      let updates = { ...input };
+      if (input.title && input.title !== existingTrack.title) {
+        const { createSlug } = await import('../utils/slug');
+        updates.slug = createSlug(input.title);
+      }
+      const updated = await updateTrack(id, updates);
+      if (!updated) {
+        throw new Error('Failed to update track');
+      }
+      return updated;
+    },
+    deleteTrack: async (_parent: any, args: any) => {
+      const { id } = args;
+      const result = await deleteTrack(id);
+      return result;
+    },
+    deleteTracks: async (_parent: any, args: any) => {
+      const { ids } = args;
+      const result = await deleteMultipleTracks(ids);
+      return result;
+    },
+    uploadTrackFile: async (_parent: any, args: any) => {
+         const { id, file } = args;
+
+         try {
+            // Await the file upload
+            const upload = await file;
+            const { filename, mimetype, createReadStream } = upload.file;
+
+            // Validate file type
+            const allowedMimeTypes = [
```

Comment:
This constant could be separated to a constants file, looks like an information that is valid for whole project, even if it's only used here.